layer/support/layer_utils.cc

```diff
@@ -15,8 +15,33 @@
 #include "layer/support/layer_utils.h"
 
 #include <cassert>
+#include <cstdint>
+
+#ifdef __linux__
+#include <sys/types.h>
+#include <unistd.h>
+#endif
 
 namespace performancelayers {
+
+int64_t GetThreadId() {
+#ifdef __linux__
+  static thread_local int64_t tid = gettid();
+  return tid;
+#else
+  return 0;
+#endif
+}
+
+int64_t GetProcessId() {
+#ifdef __linux__
+  static thread_local int64_t pid = getpid();
```

Comment:
This doesn't have to be thead-local. All threads share the same PID.

Reply:
But what if we have different processes? Will it be initialized in each process separately?

Reply:
Yes

Reply:
There's always only one process you have to worry about: the vulkan application process. If you want to coordinate across processes you need to use [some form of IPC](https://en.wikipedia.org/wiki/Inter-process_communication)

Reply:
Got it.